test/TestExtensions/MockStorageProvider.cs

```diff
@@ -45,7 +65,34 @@ public MockStorageProvider(int numKeys)
             this.numKeys = numKeys;
         }
 
-        public virtual void SetValue<TState>(string grainType, GrainReference grainReference, string name, object val)
+        public StateForTest GetProviderState()
+        {
+            var state = new StateForTest();
+            state.InitCount = initCount;
+            state.CloseCount = closeCount;
+            state.DeleteCount = deleteCount;
+            state.ReadCount = readCount;
+            state.WriteCount = writeCount;
+            return state;
+        }
+
+        [Serializable]
+        public class SetValueArgs
+        {
+            public Type StateType { get; set; }
+            public string GrainType { get; set; }
+            public GrainReference GrainReference { get; set; }
+            public string Name { get; set; }
+            public object Val { get; set; }
+
+        }
+
+        public virtual void SetValue(SetValueArgs args)
```

Comment:
not currently. I also don't understand why other method in this class is virtual. doesn't look like some class is override it. 

will remove it if not needed.